dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/utils/OptionalUtils.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.utils;
+
+import com.google.common.base.Strings;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class OptionalUtils {
+
+  public static <T> Optional<T> optionalIf(boolean condition, Supplier<T> supplier) {
```

Comment:
nit: the meaning of the name is not clear to me. I would recommend something like:

optionallyNonEmptyIf

Reply:
Renamed to `optionallyWhen` in #359 as discussed offline.